### PR TITLE
Dashboard: Rename class/feature to remove conflict with legacy dashboard widget

### LINKED ISSUE #2137: WC_Admin_Dashboard class name conflict
**Describe the bug**
In #1863 a new class was introduced: [WC_Admin_Dashboard](https://github.com/woocommerce/woocommerce-admin/blob/ea0789a59b7816ab8cf68ea19e9e7cdbd7fa5e77/includes/features/dashboard/class-wc-admin-dashboard.php) which is already a class name that is used by [WooCommerce core](https://github.com/woocommerce/woocommerce/blob/f9cdab66eb1f61f6f26001838199aa1a5b8ba01f/includes/admin/class-wc-admin-dashboard.php#L13) to build the WordPress/wp-admin Woo dashboard widget:

existing-dashboard-widget

When wc-admin is installed and active, the [class_exists check here](https://github.com/woocommerce/woocommerce/blob/f9cdab66eb1f61f6f26001838199aa1a5b8ba01f/includes/admin/class-wc-admin-dashboard.php#L13) appears to return true, and as such, the legacy core Woo dashboard widget is no longer shown.

This bug has been reported in the plugin support forums a few times: [#](https://wordpress.org/support/topic/woocommerce-dashboard-widgets-no-longer-appears/) [#](https://wordpress.org/support/topic/useless-regression/)

**To Reproduce**

1. Install WooCommerce 3.6, verify the dashboard widget is shown
2. Install wc-admin, visit the wp-admin dashboard and note that it is no longer shown.

**Expected behavior**

The legacy dashboard widget to still be available.

### PR DESCRIPTION
**Pull request by timmyc**
Fixes #2137 

This branch renames the `WC_Admin_Dashboard` to `WC_Admin_Analytics_Dashboard` to prevent the name collision that currently results in the legacy Woo Core dashboard widget from being displayed.

### Screenshots

image

### Detailed test instructions:

- On `master` of wc-admin installed/activated, note the bug by visiting the wp-admin dashboard and seeing that the WooCommerce Status widget is not shown ( see screen grab above )
- Install/npm start this branch, revisit the wp-admin dashboard and see that it is now shown
- Visit the WooCommerce Admin dashboard and verify it is still working

### CODE REVIEW

**Review by jeffstieler — APPROVED** [review]
Nice and simple! :shipit:

File: config/core.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"features": {
 		"activity-panels": false,
 		"analytics": false,
-		"dashboard": false,
+		"analytics-dashboard": false,
 		"dashboard/customizable": false,
```
Comment (justinshreve) [inline config/core.json]: Should we rename this flag to `analytics-dashboard/customizable` too?

**Comment by timmyc** [comment]
@jeffstieler is this branch still working for you? I can no longer see the dashboard on this branch - I attempted rebasing but still no love. `master` works fine.

**Comment by jeffstieler** [comment]
@timmyc I'm not seeing the dashboard anymore 😕

**Comment by psealock** [comment]
>I'm not seeing the dashboard anymore 😕

I'm not sure how this branch ever worked without updating this line too:

https://github.com/woocommerce/woocommerce-admin/blob/ea0789a59b7816ab8cf68ea19e9e7cdbd7fa5e77/client/layout/controller.js#L41

**Comment by Tobias-Conrad** [comment]
@jeffstieler @justinshreve  I was one of the admins which can not live without the WOO dashboard in the WP Dashboard.
I would like to contribute with testing, if a solution is ready please provide a download link.
At the moment it looks like, not all place of the dashboard issue have been seen ;-)
Goal could be to have this fix in the next update.

**Comment by timmyc** [comment]
Good catch @psealock guessing while testing this I had a cached build locally. I was noticing there is a lag between a file save, and changes being reflected in my local build with `npm start` running.

I applied the fix, and also changed the name of the name to `analytics-dashboard/customizable` as well per @justinshreve 's feedback.

**Review by psealock — APPROVED** [review]
LGTM

**Comment by timmyc** [comment]
> It's possible that you maybe didn't exit and restart npm start after making the change, which means the old flag would have been true still? 

Yep absolutely this was it. And that explains some of the _choice words_ I uttered while trying to get the PR working again yesterday - 😆

**Pull request merged**